demo/src/App.js

```diff
@@ -16,27 +18,229 @@ class App extends Component {
   }
 
   render() {
+    const {
+      title,
+      description,
+      children,
+      uast,
+      onNodeHover,
+      onNodeToggle
+    } = this.props;
     return (
-      <div
-        className="app"
-        style={{
-          width: '600px',
-          margin: '20px',
-          marginLeft: 'auto',
-          marginRight: 'auto'
-        }}
+      <div>
+        <h2>{title}</h2>
+        <p>{description}</p>
+        <div className="case-content">
+          <div className="sidebar">
+            <div>
+              <label>
+                <input
+                  type="checkbox"
+                  checked={this.state.showLocations}
+                  onChange={this.onShowLocation}
+                />{' '}
+                Show locations
+              </label>
+              {children}
+            </div>
+          </div>
+          <div className="viewer">
+            <UASTViewer
+              uast={uast}
+              showLocations={this.state.showLocations}
+              onNodeHover={onNodeHover}
+              onNodeToggle={onNodeToggle}
+            />
+          </div>
+        </div>
+      </div>
+    );
+  }
+}
+
+Case.propTypes = {
+  title: PropTypes.string,
+  description: PropTypes.string,
+  children: PropTypes.node,
+  uast: PropTypes.object,
+  onNodeHover: PropTypes.func,
+  onNodeToggle: PropTypes.func
+};
+
+class UncontrolledUAST extends Component {
```

Comment:
If we see the demo as a kind of documentation this file can be a bit intimidating, when the actual usage is simple. In my opinion it would be good to split this into separate files for the main app, UncontrolledUAST and ControlledUAST.

Reply:
I'm still not sure how we are going to use this demo. As in the previous PR I would suggest to leave it as is for now and improve later.
As an idea, we can convert it to use in [react storybook](https://storybook.js.org/)

Reply:
Ok. I still think is a small change that can benefit readability, please consider it in mind when you think about next steps for the demo/example code.